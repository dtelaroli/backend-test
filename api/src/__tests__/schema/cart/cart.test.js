const {
  queries: { GET_CART },
} = require("../graphql");
const { CART } = require("../../../db/seeders/20200919151449-cart");
const { seed, query } = require("../../helpers");

describe("Cart", () => {
  beforeEach(() => {
    seed("20200919151449-cart");
  });

  describe("Query", () => {
    describe("getCart", () => {
      it("Should return cart if exists", async () => {
        const result = await query({
          query: GET_CART,
          variables: { id: CART.id },
        });

        expect(result).toMatchSnapshot();
      });

      it("Should return a new cart if does not exists", async () => {
        const result = await query({
          query: GET_CART,
          variables: { id: "123" },
        });

        expect(result).toMatchSnapshot();
      });
    });
  });
});
