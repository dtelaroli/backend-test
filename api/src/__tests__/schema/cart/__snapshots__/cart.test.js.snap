// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Query getCart Should return a new cart if does not exists 1`] = `
Object {
  "data": Object {
    "getCart": Object {
      "id": "123",
      "items": Array [],
      "totalAmount": 0,
      "totalQuantity": 0,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {
        "Cache-Control": Array [
          "max-age=5, public",
        ],
      },
    },
  },
}
`;

exports[`Cart Query getCart Should return cart if exists 1`] = `
Object {
  "data": Object {
    "getCart": Object {
      "id": "j",
      "items": Array [],
      "totalAmount": 30,
      "totalQuantity": 3,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {
        "Cache-Control": Array [
          "max-age=5, public",
        ],
      },
    },
  },
}
`;
